apps/client/src/main.tsx

```diff
@@ -20,20 +21,22 @@ const root = ReactDOM.createRoot(
 root.render(
   <StrictMode>
     <Provider store={store}>
-      <BrowserRouter>
-        <Routes>
-          <Route path='/' element={<Layout />}>
-            <Route index element={<Home />} />
-            <Route path=':roomId'>
+      <ToastContextProvider>
+        <BrowserRouter>
+          <Routes>
+            <Route path='/' element={<Layout />}>
               <Route index element={<Home />} />
-              <Route path='room' element={<Room />} />
-              <Route path='race' element={<Race />} />
-              <Route path='leaderboard' element={<Leaderboard />} />
+              <Route path=':roomId'>
+                <Route index element={<Home />} />
+                <Route path='room' element={<Room />} />
+                <Route path='race' element={<Race />} />
+                <Route path='leaderboard' element={<Leaderboard />} />
+              </Route>
             </Route>
-          </Route>
-          <Route path='*' element={<NotFound />} />
-        </Routes>
-      </BrowserRouter>
+            <Route path='*' element={<NotFound />} />
+          </Routes>
+        </BrowserRouter>
+      </ToastContextProvider>
```

Comment:
@supunTE let's move the providers to a separate component in an upcoming PR no needed in this one

Reply:
Okay noted.S